src/com/itacademy/eshop/shop/ShoppingCart.java

```
@@ -19,11 +19,56 @@ public void printShoppingCart() {
         }
         System.out.println("Total price: " + getTotalPrice());
     }
+    public void addProductsToCart(ArrayList<Product> cartProducts){
+        if(!cartProducts.isEmpty()){
+            products = cartProducts;
+        } else {
+            throw new UnsupportedOperationException("Shopping cart is empty");
+        }
+    }
+    public void removeProductByName(String name) {
+        for (int i = 0; i < products.size(); i++) {
+            Product product = products.get(i);
+            if (product.getName().equals(name)) {
+                products.remove(product);
+                break;
+            }
+        }
+    }
+    public ArrayList<Product> getShoppingCartProducts(){
+        if(!products.isEmpty()){
+            return products;
+        } else {
+            throw new UnsupportedOperationException("Shopping cart is empty");
+        }
+    }
 
+    public Product findProductByName(String shirt) {
+        if(products.isEmpty()){
```

I think it could be handled better if you would throw exception if product is not found, so instead of returning null, throw exception and that would cover empty list and product not being in that list